src/SDKs/Azure.Core/data-plane/Azure.Core/RetriableStream.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    public static class RetriableStream
+    {
+        public static async Task<Stream> Create(Func<long, Task<Response>> responseFactory, int maxRetries)
```

Comment:
>public static async Task<Stream> Create(Func<long, Task<Response>> responseFactory, int maxRetries) [](start = 8, length = 99)

Should this have a mechanism to override or participate in determining which exceptions are retriable?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/5775 particulary `public abstract bool IsFatalException(Exception exception);` part of it.

Reply:
So once ResponseClassifier is available RetriableStream.Create would take a classifier parameter?

Reply:
Yes